lib/omniauth/strategies/oauth2.rb

```diff
@@ -53,14 +53,10 @@ def request_phase
       end
 
       def authorize_params
-        options.authorize_params[:state] = SecureRandom.hex(24)
-        params = options.authorize_params.merge(options_for("authorize"))
-        if OmniAuth.config.test_mode
-          @env ||= {}
-          @env["rack.session"] ||= {}
-        end
-        session["omniauth.state"] = params[:state]
-        params
+        site = self.class.name
```

Comment:
How would I add a dynamic attribute in the state params? Here is a use case: I have a multi-tenant app and in the google strategy, I have `authenticate!` method and first thing it does is switch the tenant. Since the tenant name should be dynamic, I want to get the tenant name dynamically from the state_params or from the `request.env['omniauth.auth']`. I am not sure if I am putting it correctly but I want the tenant name dynamically set somewhere in request params. If the request is coming from the` host1.com, tenant_name=host1`, `host2.com, tenant_name=host2`, something of this sort. Since the request is coming from the auth server, does it mean I need to intercept the request and add the tenant in the request params depending upon the origin?